src/Features/Lsif/GeneratorTest/SemanticTokensTests.vb

```diff
@@ -0,0 +1,84 @@
+﻿' Licensed to the .NET Foundation under one or more agreements.
+' The .NET Foundation licenses this file to you under the MIT license.
+' See the LICENSE file in the project root for more information.
+
+Imports System.Linq
+Imports Microsoft.CodeAnalysis.Editor.UnitTests.Workspaces
+Imports Microsoft.CodeAnalysis.LanguageServerIndexFormat.Generator.Graph
+Imports Microsoft.CodeAnalysis.Test.Utilities
+Imports Newtonsoft.Json
+
+Namespace Microsoft.CodeAnalysis.LanguageServerIndexFormat.Generator.UnitTests
+
+    <UseExportProvider>
+    Public NotInheritable Class SemanticTokensTests
+        Private Const TestProjectAssemblyName As String = "TestProject"
+
+        <Theory>
+        <InlineData("", "{""data"":[]}")>
+        <InlineData("namespace Foo;", "{""data"":[0,0,9,15,0,0,10,3,48,0,0,3,1,54,0]}")>
+        <InlineData("namespace " & vbCrLf & " Foo {}", "{""data"":[0,0,9,15,0,2,1,3,48,0,0,4,1,54,0,0,1,1,54,0]}")>
+        <InlineData("public class Foo { System.Collections.IList FooList { get; set; } int FooMethod() { int x = 0; return x; } }", "{""data"":[0,0,6,15,0,0,7,5,15,0,0,6,3,22,0,0,4,1,54,0,0,2,6,48,0,0,6,1,21,0,0,1,11,48,0,0,11,1,21,0,0,1,5,31,0,0,6,7,53,0,0,8,1,54,0,0,2,3,15,0,0,3,1,54,0,0,2,3,15,0,0,3,1,54,0,0,2,1,54,0,0,2,3,15,0,0,4,9,46,0,0,9,1,54,0,0,1,1,54,0,0,2,1,54,0,0,2,3,15,0,0,4,1,45,0,0,2,1,21,0,0,2,1,19,0,0,1,1,54,0,0,2,6,43,0,0,7,1,45,0,0,1,1,54,0,0,2,1,54,0,0,2,1,54,0]}")>
+        Public Async Function TestSemanticTokensData(code As String, expectedTokens As String) As Task
+            ' This test performs LSIF specific validation of the semantic tokens output. As of this writing
+            ' this feature is based on the same code path used to generate semantic tokens information in LSP
+            ' so it is assumed that exhaustive test coverage comes from that scenario.
+            '
+            ' LSIF Cases of interest:
+            ' - Empty file - generates empty data
+            ' - Single line
+            ' - Multi-line
+            ' - Roslyn custom token type ('x' in 'int x' uses a Roslyn custom token type.
+            ' - Token in last line with no trailing newline (ensure sufficient range was passed to LSP generator).
```

Comment:
Can we line these comments up with the InlineDatas?

Reply:
What do you mean? Adjust the indentation? The comments are within the function, so it seems more stylistically consistent to match the indentation of the code within, no?

Reply:
> there's five bullets here and four InlineDatas, so something seems funky

The last inline data covers both cases. The point of this comment though is to describe things that are determined specifically by the LSIF generator, not the shared SemanticTokensHelper code.

Reply:
Ah that wasn't clear at all. Can we break this into a MemberData w/ the applicable comments before each?